File: 项目/OC项目/微博(OC)/微博(OC)/Classes/Other/Category/UIBarButtonItem+Extension.m
//
//  UIBarButtonItem+Extension.m
//  微博(OC)
//
//  Created by 金亮齐 on 2017/5/11.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

#import "UIBarButtonItem+Extension.h"

@implementation UIBarButtonItem (Extension)
+ (instancetype)itemWithTarget:(id)target action:(SEL)action imageName:(NSString *)imageName highlightedImageName:(NSString *)highlightedImageName
{
    UIButton *item = [[UIButton alloc] init];
    UIImage *bgImage = [UIImage imageNamed:imageName];
    UIImage *highlightedBgImage = [UIImage imageNamed:highlightedImageName];
    [item setBackgroundImage:bgImage forState:UIControlStateNormal];
    [item setBackgroundImage:highlightedBgImage forState:UIControlStateHighlighted];
    [item addTarget:target action:action forControlEvents:UIControlEventTouchUpInside];
    item.bounds = (CGRect){CGPointZero, bgImage.size};
    return [[UIBarButtonItem alloc] initWithCustomView:item];
}

@end

File: 项目/Swift项目/微博/微博/Classes/Other/Category/UIBarButtonItem+Extension.swift
//
//  UIBarButtonItem+Extension.swift
//  微博
//
//  Created by 金亮齐 on 2017/8/29.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

import UIKit

extension UIBarButtonItem {
    
    //MARK - 抽取导航栏按钮
    class func creatBarButtonItem(imageName:String,target:AnyObject?,action:Selector) ->UIBarButtonItem {
        let btn = UIButton()
        btn.setImage(UIImage(named: imageName), for: .normal)
        btn.setImage(UIImage(named: imageName + "_highlighted"), for: .highlighted)
        btn.addTarget(target, action: action, for: .touchUpInside)
        btn.sizeToFit()
        return UIBarButtonItem(customView: btn)
    }
    
    convenience init(imageName: String, target: AnyObject?, action: String?) {
        
        let btn = UIButton()
        btn.setImage(UIImage(named: imageName), for: .normal)
        btn.setImage(UIImage(named: imageName + "_highlighted"), for: .highlighted)
        if action != nil {
            btn.addTarget(target, action: Selector(action!), for: UIControlEvents.touchUpInside)
        }
        btn.sizeToFit()
        self.init(customView: btn)
    }
}

